src/entries/popup/hooks/useKeyboardShortcut.ts

```diff
@@ -27,21 +27,12 @@ export function useKeyboardShortcut({
 }: KeyboardShortcutConfig) {
   const isCommandKVisible = useCommandKStatus((s) => s.isCommandKVisible);
 
-  // "latest ref pattern" so we don't have to worry about stabilizing the callbacks
-  const handlerRef = useRef(handler);
-  const conditionRef = useRef(condition);
-  useLayoutEffect(() => {
-    handlerRef.current = handler;
-    conditionRef.current = condition;
-  });
-
```

Comment:
Let's say our `condition` changes at some point, it might be due to a state change for example. The `useRef` will still keep the old reference to the value without it getting updated.

If you were to try to do `condition?.()` you won't get the latest value

Reply:
Same with `handler`

Reply:
can you check if this is not making it subscribe and unsubscribe from the keydown event on every render?
and if so (not sure), do you think of an alternative other than wrapping condition and handler on `useCallback` on use?
I guess the most 'performant' way would be to remember to ALWAYS call this hook like this
```ts
useKeyboardShortcut({
  handler: useCallback(() => {
    if (lalala) a()
    b()
  }, [lalala]),
  condition: useCallback(() => {
    if (lalala) return a()
    return b()
  }, [lalala])
})
```

Reply:
Yeah this will subscribe / unsubscribe when `handler` or `condition` changes. That's why we always need to use `useCallback` like you mentioned, otherwise on every re-render the function will be marked as "new / changed" everytime. Here is a good example of how `useKeyboardShortcut` should be used https://github.com/rainbow-me/browser-extension/blob/master/src/entries/popup/hooks/useTokensShortcuts.ts#L209-L212

I wonder if we should also go through all places in the code where we use `useKeyboardShortcut` and just do `useCallback` or `useMemo` for `condition` and `handler`. Maybe in the next follow up PR that'd be better ?